app/js/view/main_view.js

```diff
@@ -3,9 +3,24 @@ import 'components/gaia-header/dist/gaia-header';
 import 'components/gaia-dialog/gaia-dialog-alert';
 
 export default class MainView extends View {
-  template() {
+
+  render(isActivity) {
+    super([isActivity]);
+
+    if (isActivity) {
+      this.el.querySelector('gaia-header').addEventListener('action', event => {
+        if (event.detail.type === 'back') {
+          // Back from activity should close it via ActivityHandler.
+          window.dispatchEvent(new CustomEvent('request-activity-finish'));
+        }
+      });
+    }
+  }
+
+  template(isActivity) {
+    var action = isActivity ? `action="back"` : ``;
```

Comment:
nit: since these are single line, none templated strings let's use single quotes instead of backticks.